// @nebular/theme/components/calendar-kit/components/calendar-days-names/calendar-days-names.component.js
/**
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { NbDateTimeUtil, NbLocaleService } from '../../services';
var NbCalendarDaysNamesComponent = /** @class */ (function () {
    function NbCalendarDaysNamesComponent(locale) {
        this.locale = locale;
    }
    NbCalendarDaysNamesComponent.prototype.ngOnInit = function () {
        var days = this.createDaysNames();
        this.days = this.shiftStartOfWeek(days);
    };
    NbCalendarDaysNamesComponent.prototype.createDaysNames = function () {
        return this.locale.getDayOfWeekNames()
            .map(NbDateTimeUtil.markIfHoliday);
    };
    NbCalendarDaysNamesComponent.prototype.shiftStartOfWeek = function (days) {
        for (var i = 0; i < this.locale.getFirstDayOfWeek(); i++) {
            days.push(days.shift());
        }
        return days;
    };
    NbCalendarDaysNamesComponent.decorators = [
        { type: Component, args: [{
                    selector: 'nb-calendar-days-names',
                    styles: [":host{display:flex;justify-content:space-between}:host .day{display:flex;align-items:center;justify-content:center;margin:1px} "],
                    template: "\n    <div class=\"day\" *ngFor=\"let day of days\" [class.holiday]=\"day.isHoliday\">{{ day.name }}</div>\n  ",
                    changeDetection: ChangeDetectionStrategy.OnPush,
                },] },
    ];
    /** @nocollapse */
    NbCalendarDaysNamesComponent.ctorParameters = function () { return [
        { type: NbLocaleService, },
    ]; };
    return NbCalendarDaysNamesComponent;
}());
export { NbCalendarDaysNamesComponent };
