// @nebular/theme/components/calendar-kit/services/locale.service.js
/**
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */
import { Inject, Injectable, LOCALE_ID } from '@angular/core';
import { FormStyle, getLocaleDayNames, getLocaleFirstDayOfWeek, getLocaleMonthNames, TranslationWidth, } from '@angular/common';
var NbLocaleService = /** @class */ (function () {
    function NbLocaleService(locale) {
        this.locale = locale;
    }
    /**
     * returns first day of the week, it can be 1 if week starts from monday
     * and 0 if from sunday and so on.
     * */
    /**
       * returns first day of the week, it can be 1 if week starts from monday
       * and 0 if from sunday and so on.
       * */
    NbLocaleService.prototype.getFirstDayOfWeek = /**
       * returns first day of the week, it can be 1 if week starts from monday
       * and 0 if from sunday and so on.
       * */
    function () {
        return getLocaleFirstDayOfWeek(this.locale);
    };
    NbLocaleService.prototype.getMonthName = function (date, style) {
        if (style === void 0) { style = TranslationWidth.Abbreviated; }
        var index = date.getMonth();
        return this.getMonthNameByIndex(index, style);
    };
    NbLocaleService.prototype.getMonthNameByIndex = function (index, style) {
        if (style === void 0) { style = TranslationWidth.Abbreviated; }
        return getLocaleMonthNames(this.locale, FormStyle.Format, style)[index];
    };
    NbLocaleService.prototype.getDayOfWeekNames = function () {
        return getLocaleDayNames(this.locale, FormStyle.Format, TranslationWidth.Narrow);
    };
    NbLocaleService.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    NbLocaleService.ctorParameters = function () { return [
        { type: undefined, decorators: [{ type: Inject, args: [LOCALE_ID,] },] },
    ]; };
    return NbLocaleService;
}());
export { NbLocaleService };
